source/extensions/tracers/opencensus/opencensus_tracer_impl.cc

```diff
@@ -0,0 +1,171 @@
+#include "extensions/tracers/opencensus/opencensus_tracer_impl.h"
+
+#include "absl/strings/str_cat.h"
+#include "opencensus/exporters/trace/stackdriver/stackdriver_exporter.h"
+#include "opencensus/exporters/trace/stdout/stdout_exporter.h"
+#include "opencensus/exporters/trace/zipkin/zipkin_exporter.h"
+#include "opencensus/trace/propagation/cloud_trace_context.h"
+#include "opencensus/trace/propagation/grpc_trace_bin.h"
+#include "opencensus/trace/propagation/trace_context.h"
+#include "opencensus/trace/sampler.h"
+#include "opencensus/trace/span.h"
+#include "opencensus/trace/trace_config.h"
+#include "opencensus/trace/trace_params.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace OpenCensus {
+
+namespace {
+
+/**
+ * OpenCensus tracing implementation of the Envoy Span object.
+ */
+class Span : public Tracing::Span {
+public:
+  Span(const Tracing::Config& config, const envoy::config::trace::v2::OpenCensusConfig* oc_config,
+       Http::HeaderMap& request_headers, const std::string& operation_name, SystemTime start_time,
+       const Tracing::Decision tracing_decision);
+
+  // Used by spawnChild().
+  Span(const envoy::config::trace::v2::OpenCensusConfig* oc_config,
+       ::opencensus::trace::Span&& span);
+
+  void setOperation(const std::string& operation) override;
+  void setTag(const std::string& name, const std::string& value) override;
+  void finishSpan() override;
+  void injectContext(Http::HeaderMap& request_headers) override;
+  Tracing::SpanPtr spawnChild(const Tracing::Config& config, const std::string& name,
+                              SystemTime start_time) override;
+  void setSampled(bool sampled) override;
+
+private:
+  // TODO: Do we take the sampling decision from StartSpan, or use the sampler
+  // configured in TraceConfig?
```

Comment:
The approach taken in zipkin tracer is, if the trace context passed with the request identifies a previous sampling decision for the trace instance, then use that - otherwise use the decision passed in by envoy. However that doesn't necessarily mean all tracers need to follow the same approach.